src/rust/catloop/mod.rs

```diff
@@ -249,6 +259,7 @@ impl SharedCatloopLibOS {
         match result {
             Ok(new_queue) => {
                 let new_qd: QDesc = self.runtime.alloc_queue::<SharedCatloopQueue>(new_queue);
+                // TODO: insert into socket id to queue descriptor table?
```

Comment:
Which scenario would be prevented if this is done? Duplicate entries, like on line 201?

Reply:
If the application tries to later bind to an ephemeral port or some other port that has already been used for accepting or connecting, we should not allow that. However, for Catloop, I don't think connections have address pairs, they just use the server-side address and port (e.g., on the client, the queue only has a remote address and on the server, the queue only has a local address), so it's unclear what the right behavior is here.